canbench-rs/src/lib.rs

```diff
@@ -648,20 +641,71 @@ fn reset() {
     SCOPES.with(|p| p.borrow_mut().clear());
 }
 
-// Returns the measurements for any declared scopes,
-// aggregated by the scope name.
-fn get_scopes_measurements() -> std::collections::BTreeMap<&'static str, Measurement> {
-    SCOPES
-        .with(|p| p.borrow().clone())
-        .into_iter()
-        .map(|(scope, measurements)| {
-            let mut total = Measurement::default();
-            for measurement in measurements {
-                total = total + measurement;
+// Returns the measurements for any declared scopes, aggregated by the scope name.
+fn get_scopes_measurements() -> BTreeMap<&'static str, Measurement> {
```

Comment:
nit: not sure if the entire logic is covered with the tests.

Reply:
I'll keep it as is, let me know if there are specific test cases that should be added.